File: client/src/app/shared/Student.ts
import { ICourse } from "./Course";
import { IStudentCourse } from "./StudentCourse";
import { ISubjectGrade } from "./SubjectGrade";

export class IStudent {
    studentId: number;
    firstName: string;
    lastName: string;
    age: number;
    studentCourse: IStudentCourse[];
    subjectGrade: ISubjectGrade[]
}

File: Data/Entities/Student.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Magnifinance.Data.Entities
{
    public class Student
    {
        public int StudentId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }

        public DateTime EnrollmentDate { get; set; }
        public int Age { get; set; }
        //public List<Course> StudentCourse { get; set; }


        public List<StudentCourse> StudentCourse { get; set; }

        public List<SubjectGrade> SubjectGrade { get; set; }

    }

}
